apps/src/code-studio/components/progress/TeacherUnitOverview.tsx

```diff
@@ -239,6 +240,16 @@ const initializeRedux = (
     dispatch(setStudentDefaultsSummaryView(false));
   }
 
+  if (unitData.showCalendar) {
+    console.log('inside showCalendar');
+    dispatch(setShowCalendar(true));
+  }
+
+  if (unitData.calendarLessons) {
+    console.log('inside calendarLessons');
+    dispatch(setCalendarLessons(unitData.calendarLessons));
```

Comment:
Same here, we should probably set this to an empty value if there are no calendar lessons.